src/status_im/contexts/onboarding/create_password/view.cljs

```diff
@@ -119,102 +130,70 @@
-      [rn/view
-       {:style style/footer-container}
-       (when same-passwords?
-         [rn/view {:style style/disclaimer-container}
-          [quo/disclaimer
-           {:blur?               true
-            :on-change           (partial set-accepts-disclaimer? not)
-            :checked?            accepts-disclaimer?
-            :customization-color user-color}
-           (i18n/label :t/password-creation-disclaimer)]])
-       (when (and (= focused-input :password) (not same-passwords?))
-         [help
-          {:validations       password-validations
-           :password-strength password-strength}])
-       [quo/button
-        {:container-style     style/footer-button-container
-         :disabled?           (not meet-requirements?)
-         :customization-color user-color
-         :on-press            #(rf/dispatch
-                                [:onboarding/password-set
-                                 (security/mask-data password)])}
-        (i18n/label :t/password-creation-confirm)]]}
-
+      :blur-options                         style/blur-options
+      :footer-container-padding             0
+      :footer                               [help-and-confirm-button
+                                             {:password-validations password-validations
+                                              :same-passwords?      same-passwords?
+                                              :on-submit            on-submit}]}
      [rn/view {:style style/form-container}
       [header]
       [password-inputs
-       {:password-long-enough?     password-long-enough?
-        :password-short-enough?    password-short-enough?
-        :passwords-match?          same-passwords?
-        :empty-password?           empty-password?
-        :show-password-validation? show-password-validation?
-        :on-input-focus            #(set-focused-input :password)
-        :on-change-password        (fn [new-value]
-                                     (set-password new-value)
-                                     (when same-password-length?
-                                       (set-show-password-validation? true)))
-
-        :on-change-repeat-password (fn [new-value]
-                                     (set-repeat-password new-value)
-                                     (when same-password-length?
-                                       (set-show-password-validation? true)))
-        :on-blur-repeat-password   #(if empty-password?
-                                      (set-show-password-validation? false)
-                                      (set-show-password-validation? true))}]]]))
+       {:password-long-enough?  long-enough?
+        :password-short-enough? short-enough?
+        :non-empty-password?    non-empty?
+        :same-passwords?        same-passwords?
+        :same-password-length?  same-password-length?
+        :set-password           set-password
+        :set-repeat-password    set-repeat-password}]]]))
```

Comment:
Along with the improvement about caching some callbacks, I simplified this screen too. It musts behave as before, this is just a refactor. 

CC: @ilmotta